File: leetcode/encode-and-decode-tinyurl.py
class Codec:
    
    def __init__(self):
        self.url_pair = {}
    
    def encode(self, longUrl):
        """Encodes a URL to a shortened URL.
        
        :type longUrl: str
        :rtype: str
        """
        suffix = ''.join([random.choice(string.ascii_letters + string.digits) for i in range(6)])
        shortUrl = 'http://tinyurl.com/' + suffix
        self.url_pair[shortUrl] = longUrl
        return shortUrl
        

    def decode(self, shortUrl):
        """Decodes a shortened URL to its original URL.
        
        :type shortUrl: str
        :rtype: str
        """
        return self.url_pair.get(shortUrl, 'Not an encodes TinyUrl')
        

# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(url))

File: leetcode-questions/encode-and-decode-tinyurl.cpp
class Solution {
public:
    map<string, string> url_pair;
    string ascii = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
    // Encodes a URL to a shortened URL.
    string encode(string longUrl) {
        string suffix = "";
        int n = ascii.size();
        for(int i=0;i<6;i++) {
            int j = rand()%n;
            char c = ascii[j];
            suffix += c;
        }
        string shortUrl = "http://tinyurl.com/" + suffix;
        url_pair[shortUrl] = longUrl;
        return shortUrl;
    }

    // Decodes a shortened URL to its original URL.
    string decode(string shortUrl) {
        return url_pair[shortUrl];
    }
};

// Your Solution object will be instantiated and called as such:
// Solution solution;
// solution.decode(solution.encode(url));